mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -349,13 +349,13 @@ def _serve(model_uri, port, host):
 
 
 def get_cmd(
-    model_uri: str, port: int = None, host: int = None, nworkers: int = None
+    model_uri: str, port: int = None, host: int = None, timeout: int = None, nworkers: int = None
 ) -> Tuple[str, Dict[str, str]]:
     local_uri = path_to_local_file_uri(model_uri)
     # NB: Absolute windows paths do not work with mlflow apis, use file uri to ensure
     # platform compatibility.
     if os.name != "nt":
-        args = ["--timeout=60"]
```

Comment:
Curious what would happen if we specify timeout using `--gunicorn-opts` argument?

```
mlflow models serve --gunicorn-opts='--timeout=xxx'
```

Reply:
it would still work. I noticed this gunicorn functionality as well.
However imho, a dedicated argument is always a good idea, isn't ?

Additionally, there is an issue concerning dockerized version of scoring server. And that is the usage of different worker class (`gevent`)  where `timeout` has not effect there. [source](https://githubhot.com/repo/benoitc/gunicorn/issues/2695)
https://github.com/mlflow/mlflow/blob/55d176f5b7def30486f60f59598d7431ea8e7b69/mlflow/models/docker_utils.py#L33-L40

Reply:
It does only start `_serve()` method
https://github.com/mlflow/mlflow/blob/101ad6e8eb383c769178df0df83d1d2a1cea6b4a/mlflow/pyfunc/backend.py#L248

https://github.com/mlflow/mlflow/blob/101ad6e8eb383c769178df0df83d1d2a1cea6b4a/mlflow/models/container/__init__.py#L64-L70

Either way, `-k gevent` argument, cancels out `--timeout` and I'm curious that there is difference in gunicorn options between
 - local deployments (no gunircorn option is set, `sync` is the default) and 
 - dockerized deployments (`-k gevent` explicitly defined)
 
 I think we can probably address the issue in another PR and then provide `timeout` option for `mlflow models build-docker`

Reply:
I agree :)

Reply:
@harupy i think that failed tests are also failing also in project's master branch when I am testing locally. Can you confirm this?

Reply:
Are you sure? Have you seen this error?

https://github.com/mlflow/mlflow/runs/6016042645?check_suite_focus=true#step:9:286